collections/ansible_collections/massopencloud/esi/roles/network/tasks/destroy_network.yaml

```
@@ -0,0 +1,7 @@
+- name: Check that exactly one network matching the ident exists  # noqa:no-changed-when
+  ansible.builtin.command: >-
+    openstack network show {{ network_identifier }} -f json
+
+- name: Destroy network  # noqa:no-changed-when
+  ansible.builtin.command: >-
+    openstack network delete {{ network_identifier }}
```

```suggestion
    openstack.cloud.network:
      name: "{{ network_name }}"
      state: absent 
```
looks like it's possible: https://docs.ansible.com/ansible/latest/collections/openstack/cloud/network_module.html#parameter-state

---

Oh, I tried it, based on that documentation; it doesn't actually seem to work if you use the UUID rather than the name, unfortunately.

---

https://opendev.org/openstack/ansible-collections-openstack/src/branch/master/plugins/modules/network.py#L241

the module does actually find the network by name, before using its id later to perform the deletion.

---

I think it makes sense to require a name when creating a network; as it is right now, it will accept either a name or a UUID when deleting, as `show` accepts either. It seems like a reasonable compromise to me, but I'm open to other behaviors!

---

It's weird because the underlying function should take both id or name:
https://opendev.org/openstack/openstacksdk/src/branch/master/openstack/network/v2/_proxy.py#L2400-L2418

---

It is really strange - I double-checked just now, and can confirm that for whatever reason the openstack network module doesn't work with a ID. I've made the other changes; my reasoning for not grabbing the ID was that the command proves that there is only one matching network, but it is much safer to use the ID as you suggested. Let me know if you have any other questions or concerns!